Extension/src/LanguageServer/configurations.ts

```diff
@@ -1536,9 +1536,15 @@ export class CppProperties {
 
         // Check if config name is unique.
         errors.name = this.isConfigNameUnique(config.name);
-
+        let resolvedCompilerPath: string | undefined;
         // Validate compilerPath
-        let resolvedCompilerPath: string | undefined = this.resolvePath(config.compilerPath);
+        if (config.compilerPath) {
+            resolvedCompilerPath = which.sync(config.compilerPath, { nothrow: true }) ?? undefined;
```

Comment:
Not sure we want to just randomly call `which` on the compilerPath without checking if it looks like a path first.  Should we be checking for path separators first?

Reply:
Looking at [the code to `which`](https://www.npmjs.com/package/which?activeTab=code), it does appear to do the right thing (not resolve against paths) if provided input with a path separator in it.  Though, I do notice that it also considers the `cwd`, which may not always be desirable (as it's unclear what we can expect the cwd to be at any given time - likely not always the current workspace folder, i.e. in multiroot).